src/pages/Search/SearchFiltersCategoryPage.tsx

```diff
@@ -0,0 +1,96 @@
+import React, {useMemo} from 'react';
+import {View} from 'react-native';
+import {useOnyx} from 'react-native-onyx';
+import FullPageNotFoundView from '@components/BlockingViews/FullPageNotFoundView';
+import type {FormOnyxValues} from '@components/Form/types';
+import HeaderWithBackButton from '@components/HeaderWithBackButton';
+import ScreenWrapper from '@components/ScreenWrapper';
+import SelectionList from '@components/SelectionList';
+import RadioListItem from '@components/SelectionList/RadioListItem';
+import useLocalize from '@hooks/useLocalize';
+import useThemeStyles from '@hooks/useThemeStyles';
+import localeCompare from '@libs/LocaleCompare';
+import * as SearchActions from '@userActions/Search';
+import ONYXKEYS from '@src/ONYXKEYS';
+
+function SearchFiltersStatusPage() {
+    const styles = useThemeStyles();
+    const {translate} = useLocalize();
+
+    const [searchAdvancedFiltersForm] = useOnyx(ONYXKEYS.FORMS.SEARCH_ADVANCED_FILTERS_FORM);
+    const activeItems = searchAdvancedFiltersForm?.category;
+    const policyId = searchAdvancedFiltersForm?.policyId ?? '-1';
+
+    const [callCategories] = useOnyx(ONYXKEYS.COLLECTION.POLICY_CATEGORIES);
+    const [singlePolicyCategories] = useOnyx(`${ONYXKEYS.COLLECTION.POLICY_CATEGORIES}${policyId}`);
```

Comment:
@blazejkustra need your input on this
we need to have both:
 - all the categories available in onyx in case `policyID` is `undefined`
 - only the categories for a specific policy in case `policyID` is a correct id

Is this approach correct? Im asking mostly about this line: `useOnyx(ONYXKEYS.COLLECTION.POLICY_CATEGORIES)` - Im not sure if it's legit to connect to the whole COLLECTION.

Reply:
AFAIK you can connect to the whole collection, yes. But if `policyID` is undefined, wouldn't we connect to `policyCategories_undefined` in this case? Maybe we need a ternary? 

```
const [singlePolicyCategories] = useOnyx(policyID ? `${ONYXKEYS.COLLECTION.POLICY_CATEGORIES}${policyId}` : ONYXKEYS.COLLECTION.POLICY_CATEGORIES);
```

Reply:
I think the correct way of doing this is to have a fallback value:
```ts
    const [singlePolicyCategories] = useOnyx(`${ONYXKEYS.COLLECTION.POLICY_CATEGORIES}${policyId ?? '-1'}`);
```

From STYLE.md:
```
### Default value for inexistent IDs

 Use `'-1'` or `-1` when there is a possibility that the ID property of an Onyx value could be `null` or `undefined`.

``` ts
// BAD
const foo = report?.reportID ?? '';
const bar = report?.reportID ?? '0';

report ? report.reportID : '0';
report ? report.reportID : '';

// GOOD
const foo = report?.reportID ?? '-1';

report ? report.reportID : '-1';
```

Reply:
Would that connect to the whole collection though?

Reply:
Oh right, that's for the case where policyID is undefined (then the rule is to use -1)

If we need the whole collection AND a single policy' categories then what we could utilize is:
```ts
    const [policyCategories] = useOnyx(ONYXKEYS.COLLECTION.POLICY_CATEGORIES);
const singlePolicyCategory = POLICY_CATEGORIES[policyID ?? "-1"]
```

Would that work for your use case @289Adam289 @Kicu?

Reply:
Yea, seems similar to my suggestion [above](https://github.com/Expensify/App/pull/46197#discussion_r1693257733)

Reply:
The only drawback of your approach is a more complex type that might prove hard to use:
```ts
singlePolicyCategories: PolicyCategories | Record<string, PolicyCategories | undefined> | undefined
```

But yes, these are similar 👍